transforms/pair.go

```diff
@@ -0,0 +1,137 @@
+package transforms
+
+import (
+	"github.com/facebookgo/errgroup"
+	"gopkg.in/Clever/optimus.v3"
+)
+
+// PairType is the type of join to use when Pairing
+type PairType int
+
+// RowHasher takes in a row and returns a hash for that Row.
```

Comment:
nit: i would name this something other than hasher, given that it doesn't necessarily need to do any hashing. I looked up what the call this, and the wikipedia article for hash join calls them "join attributes" so maybe RowAttribute? what do you think?